sirepo/pkcli/job_supervisor.py

```diff
@@ -106,12 +114,49 @@ def sr_on_exception(self):
         self.close()
 
 
+# Must be defined before use
+def _db_file_req_validated(func):
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        t = self.request.headers.get('Authorization')
+        if not t:
+            raise tornado.web.HTTPError(401)
+        p = t.split(' ')
+        if len(p) != 2:
+            raise tornado.web.HTTPError(401)
+        if p[0] != 'Bearer' or not sirepo.job.UNIQUE_KEY_RE.search(p[1]):
+            raise tornado.web.HTTPError(401)
+        if p[1] not in _DB_FILE_KEYS or \
+           self.request.path.split('/')[3] != _DB_FILE_KEYS[p[1]]:
+            raise tornado.web.HTTPError(403)
+        # TODO(e-carlin): discuss with rn what validation needs to happen
```

Comment:
@robnagler how should we validate the path?

Reply:
https://github.com/radiasoft/sirepo/pull/3266/commits/b3499213ed568ed41b026c8d6018030c5d47eec2